### PR TITLE
Upgrade to v2.5.0

### PR DESCRIPTION
**Pull request by metinavc1**

### CODE REVIEW

**Review by chrishulbert — APPROVED** [review]

File: Library/Model/SPIPayAtTable.h [hunk 1]
```diff
@@ -82,9 +84,29 @@ typedef NS_ENUM(NSUInteger, SPIPaymentType) {
 
 @end
 
-@interface SPIBillPayment : NSObject
+@interface SPIOpenTablesEntry : NSObject
 
-- (instancetype)initWithMessage:(SPIMessage *)message;
+@property (nonatomic, retain) NSString *tableId;
```
Comment (chrishulbert) [inline Library/Model/SPIPayAtTable.h]:
```suggestion
@property (nonatomic, copy) NSString *tableId;
```

File: Library/Model/SPIPayAtTable.h [hunk 2]
```diff
@@ -82,9 +84,29 @@ typedef NS_ENUM(NSUInteger, SPIPaymentType) {
...
+@property (nonatomic, retain) NSString *label;
```
Comment (chrishulbert) [inline Library/Model/SPIPayAtTable.h]:
```suggestion
@property (nonatomic, copy) NSString *label;
```

File: Library/Model/SPIPayAtTable.h [hunk 3]
```diff
@@ -82,9 +84,29 @@ typedef NS_ENUM(NSUInteger, SPIPaymentType) {
...
+
+- (instancetype)initWithDictionary:(NSDictionary *)data;
+
+- (NSDictionary *)toJsonObject;
+
+@end
+
+@interface SPIGetOpenTablesResponse : NSObject
+
+@property (nonatomic, retain) NSString *tableData;
```
Comment (chrishulbert) [inline Library/Model/SPIPayAtTable.h]:
```suggestion
@property (nonatomic, copy) NSString *tableData;
```

File: Library/Model/SPIPayAtTable.h [hunk 4]
```diff
@@ -143,4 +173,24 @@ typedef NS_ENUM(NSUInteger, SPIPaymentType) {
...
 
+- (void)handleBillPaymentFlowEnded:(SPIMessage *)message;
+
+- (void)handleGetOpenTablesRequest:(SPIMessage *)message;
+
+@end
+
+@interface SPIBillPaymentFlowEndedResponse : NSObject
+
+@property (nonatomic, retain) NSString *billId;
```
Comment (chrishulbert) [inline Library/Model/SPIPayAtTable.h]:
```suggestion
@property (nonatomic, copy) NSString *billId;
```

File: Library/Model/SPITransaction.m [hunk 5]
```diff
@@ -585,8 +585,9 @@ - (instancetype)initWithAmountCents:(NSInteger)amountCents
 - (SPIMessage *)toMessage {
     NSMutableDictionary *data = [[NSMutableDictionary alloc] init];
     [data setValue:_posRefId forKey:@"pos_ref_id"];
-    [data setValue:[NSNumber numberWithInteger:_purchaseAmount] forKey:@"purchase_amount"];
-    [data setValue:[NSNumber numberWithInteger:_surchargeAmount] forKey:@"surcharge_amount"];
+    [data setValue:@(_purchaseAmount) forKey:@"purchase_amount"];
```
Comment (chrishulbert) [inline Library/Model/SPITransaction.m:589]: neat!

**Review by chrishulbert — CHANGES_REQUESTED** [review]
Looks pretty good!

File: Library/Model/SPIPayAtTable.h (hunk 1 above)
Comment (metinavc1) [reply Library/Model/SPIPayAtTable.h]: done

File: Library/Model/SPIPayAtTable.h (hunk 2 above)
Comment (metinavc1) [reply Library/Model/SPIPayAtTable.h]: done

File: Library/Model/SPIPayAtTable.h (hunk 3 above)
Comment (metinavc1) [reply Library/Model/SPIPayAtTable.h]: done

File: Library/Model/SPIPayAtTable.h (hunk 4 above)
Comment (metinavc1) [reply Library/Model/SPIPayAtTable.h]: done

**Review by chrishulbert — APPROVED** [review]

**Review by chrishulbert — APPROVED** [review]
I think i reviewed all this already in another PR

**Review by chrishulbert — APPROVED** [review]

**Pull request merged**